webhook/resourcesemantics/validation/validation.go

```diff
@@ -46,11 +48,15 @@ import (
 
 var errMissingNewObject = errors.New("the new object may not be nil")
 
+// Callback is a generic function to be called by a consumer of validation
+type Callback func(context.Context, unstructured.Unstructured) error
```

Comment:
Should this take ptr to Unstructured (the contents is a map, which a referenced object, so 🤷‍♂), since you can't do much with actual object (all its methods require pointer receiver).

Reply:
Sure. I'm not sure if it's a big difference, but let's go with the pointer.